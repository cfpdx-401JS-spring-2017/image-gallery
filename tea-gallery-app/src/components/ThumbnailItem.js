import React from 'react';
import PropTypes from 'prop-types';
import { DeleteImageButton } from './DeleteButtons';

ThumbnailItem.PropTypes = {
  images: PropTypes.array.required,
  onDeleteImage: PropTypes.func.required
};

export default function ThumbnailItem(props) {

  if(!props.images) return <div>No images in this album, add some below</div>;
  return (
    <ul className="thumbnail-list">
      {props.images.map((image, i) => (
        <div key={i} className="thumbnail-wrapper">
          <li className="thumbnail-li">
            <img className="thumbnail" src={image.url} alt="tea"/>
          </li>
          <DeleteImageButton 
          images={props.images}
          onDeleteImage={() => props.onDeleteImage(image._id)}
          />
        {props.images.length < 1 && <h2>Album is empty. Add images below</h2>}
        </div>
      ))}
    </ul>
  );
}