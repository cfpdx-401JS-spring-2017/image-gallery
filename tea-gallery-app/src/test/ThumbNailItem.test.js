import React from 'react';
import renderer from 'react-test-renderer';
import ThumbnailItem from '../components/ThumbnailItem';

const images = [
  {
    title: 'tea leaves in teapot',
    description: 'what nice leaves',
    url: '/../public/img/teainpot.jpg'
  },
  {
    title: 'Tetsubin over charcoal',
    description: 'what a nice tetsubin',
    url: '/../public/img/tetsubin-charcoal.jpg'
  },
  {
    title: 'Teapot display',
    description: 'what a nice teapot display',
    url: '/../public/img/teapot-display.jpg'
  }
];

describe('ThumbnailItem', () => {
  it('renders the ThumbnailItem component', () => {
    const component = renderer.create(<ThumbnailItem images={images}/>);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

});