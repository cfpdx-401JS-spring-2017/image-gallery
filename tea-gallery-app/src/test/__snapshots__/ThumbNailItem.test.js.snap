// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThumbnailItem renders the ThumbnailItem component 1`] = `
<ul
  className="thumbnail-list"
>
  <div
    className="thumbnail-wrapper"
  >
    <li
      className="thumbnail-li"
    >
      <img
        alt="tea"
        className="thumbnail"
        src="/../public/img/teainpot.jpg"
      />
    </li>
    <button
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
  <div
    className="thumbnail-wrapper"
  >
    <li
      className="thumbnail-li"
    >
      <img
        alt="tea"
        className="thumbnail"
        src="/../public/img/tetsubin-charcoal.jpg"
      />
    </li>
    <button
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
  <div
    className="thumbnail-wrapper"
  >
    <li
      className="thumbnail-li"
    >
      <img
        alt="tea"
        className="thumbnail"
        src="/../public/img/teapot-display.jpg"
      />
    </li>
    <button
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</ul>
`;
